src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py

```diff
@@ -138,134 +138,115 @@ def evaluate(
         operational_settings_used = TimeSeriesInt(
             timesteps=variables_map.time_vector, values=[0] * len(variables_map.time_vector), unit=Unit.NONE
         )
-        operational_settings_results: Dict[int, List[ConsumerModelResult]] = defaultdict(list)
+        operational_settings_results: Dict[int, Dict[str, EcalcModelResult]] = defaultdict(
+            dict
+        )  # map operational settings index and consumer id to consumer result.
 
         for period, operational_settings in temporal_operational_settings.items():
             variables_map_for_period = variables_map.get_subset_from_period(period)
             start_index, end_index = period.get_timestep_indices(variables_map.time_vector)
             for operational_setting_index, operational_setting in enumerate(operational_settings):
-                operational_setting_result = self._evaluate_operational_setting(
-                    operational_setting=operational_setting, variables_map=variables_map_for_period
+                adjusted_operational_settings: List[
+                    ConsumerOperationalSettings
+                ] = self._get_operational_settings_adjusted_for_crossover(
+                    operational_setting=operational_setting,
+                    variables_map=variables_map_for_period,
+                )
+
+                for consumer, adjusted_operational_setting in zip(self._consumers, adjusted_operational_settings):
+                    consumer_result = consumer.evaluate(adjusted_operational_setting)
+                    operational_settings_results[operational_setting_index][consumer.id] = consumer_result
+
+                consumer_results = operational_settings_results[operational_setting_index].values()
+
+                # Check if consumers are valid for this operational setting, should be valid for all consumers
+                is_all_consumers_valid = reduce(
```

Comment:
is it the consumer that is valid or the results? rename to `all_consumer_results_valid`? prefix `is_` is reserved for functions?